beacon-chain/db/iface/interface.go

```diff
@@ -108,6 +108,9 @@ type NoHeadAccessDatabase interface {
 	SaveSlasherChunks(
 		ctx context.Context, kind slashertypes.ChunkKind, chunkKeys []uint64, chunks [][]uint16,
 	) error
+	CheckAndUpdateForSlashableProposals(
```

Comment:
This is a DB interface. I am not sure that it should be responsible for checking/verifying things.